libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat

```diff
@@ -0,0 +1,240 @@
+# hw definition file for processing by chibios_hwdef.py for MFT-SEMA100
+
+# MCU class and specific type
+MCU STM32H7xx STM32H743xx
+
+# crystal frequency
+OSCILLATOR_HZ 16000000
+
+# board ID for firmware load
+APJ_BOARD_ID 1500
+
+FLASH_SIZE_KB 2048
+
+# with 2M flash we can afford to optimize for speed
+env OPTIMIZE -O2
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 2
+
+
+
+USB_STRING_MANUFACTURER "MFT"
+USB_STRING_PRODUCT "MFT-SEMA100"
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART1 USART3 UART5 UART7 UART8
+
+# now we define the pins that USB is connected on
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# Port switching for USB HS and FS,high = USB_FS , LOW = USB_HS
+PH15 USB_HS_ENABLE OUTPUT HIGH
+define USB_HW_ENABLE_HS 0
+
+# these are the pins for SWD debugging with a STlinkv2 or black-magic probe
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# SPI1 - SD CARD
+PB3 SPI1_SCK SPI1
+PB4 SPI1_MISO SPI1
+PD7 SPI1_MOSI SPI1
+PA15 SDCARD_CS CS
+PC4 SDCARD_DETECT INPUT
+
+# SPI2 - External 
+PD3 SPI2_SCK SPI2
+PC2 SPI2_MISO SPI2
+PC3 SPI2_MOSI SPI2
+
+
+# SPI4 - GYRO
+PE2 SPI4_SCK SPI4
+PE5 SPI4_MISO SPI4
+PE6 SPI4_MOSI SPI4
+
+# sensor
+PE3  BMI088_A_CS CS
+PE4  BMI088_G_CS CS
+
+
+# I2C buses
+
+# I2C1 is External
+PB8 I2C1_SCL I2C1
+PB7 I2C1_SDA I2C1
+
+# I2C2 for onboard mag and External
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+# I2C3 for onboard BAROMETER
+PA8 I2C3_SCL I2C3
+PC9 I2C3_SDA I2C3
+
+# I2C4 is External
+PD12 I2C4_SCL I2C4
+PD13 I2C4_SDA I2C4
+
+# order of I2C buses
+I2C_ORDER I2C2 I2C3 I2C4 I2C1
+
+
+define HAL_I2C_INTERNAL_MASK 1
```

Comment:
this should be a mask of what buses are internal only. looks like only I2C3 is internal-only? If that is correct then a mask of 2 would be right